src/installer/installer.go

```diff
@@ -193,23 +200,44 @@ func (i *installer) waitForControlPlane(ctx context.Context, kc k8s_client.K8SCl
 		i.log.Errorf("Timeout waiting for master nodes, %s", err)
 		return err
 	}
+
 	if err := kc.PatchEtcd(); err != nil {
 		i.log.Error(err)
 		return err
 	}
+
+	err := i.addPolicyForController()
+	if err != nil {
+		i.log.Error(err)
+		return err
+	}
+
 	i.waitForBootkube(ctx)
+
 	return nil
 }
 
 func (i *installer) UpdateHostStatus(newStatus string) {
 	i.log.Infof("Updating node installation status: %s", newStatus)
 	if i.HostID != "" {
-		if err := i.ic.UpdateHostStatus(newStatus); err != nil {
+		if err := i.ic.UpdateHostStatus(newStatus, i.HostID); err != nil {
 			i.log.Errorf("Failed to update node installation status, %s", err)
 		}
 	}
 }
 
+func (i *installer) addPolicyForController() error {
+	i.log.Infof("Adding policy for assisted-deployment namespace")
+	err := retry.Do(
+		func() error {
+			_, err := i.ops.ExecPrivilegeCommand("oc", fmt.Sprintf("--kubeconfig=%s", KubeconfigPath),
```

Comment:
i failed to find a way to run this with a client. Will be glad for help. We can open a ticket to move it to client.